### ISSUE TITLE
Allow changing window dimensions config options without a restart

### ISSUE THREAD

**Issue by eikenb** [alacritty/alacritty#8200]
Documentation states `Window dimensions (changes require restart).` 
This also means using `alacritty msg config` doesn't work.

I'd like to be able to change the window size at runtime using `alacritty msg config`. I currently have to use wmctrl which takes the new window size in pixels and this makes it difficult to change as it requires cell sizes to be calculated the changes and cell sizes change if I adjust the font size.

TLDR; I'd like (eg.) `alacritty msg config window.dimentions.lines=50` to work.

Thanks.

**Comment by chrisduerr** [comment]
I can see how this might make sense for IPC, but the config option alone is a little questionable. So this likely isn't going to happen unless you're willing to send a patch yourself.

**Comment by sethm** [comment]
I'm kind of in the same boat, where I'd like to be able to adjust the dimensions (both `columns` and `lines`) at runtime. My ulterior motive is that I'd like `IncreaseFontSize` to increase the font size while maintaining the dimensions. I haven't dug into the code deeply yet, but maybe I can come up with a PR to help both @eikenb and myself out.

**Comment by chrisduerr** [comment]
@sethm I haven't looked at the code myself, so use at your own risk, but you might be interested in https://github.com/alacritty/alacritty/pull/8560.
